compio-driver/src/unix/op.rs

```diff
@@ -370,6 +370,105 @@ impl<T: IoVectoredBuf, S> IntoInner for SendVectored<T, S> {
     }
 }
 
+/// Receive data and source address with ancillary data into vectored buffer.
+pub struct RecvMsg<T: IoVectoredBufMut, C: IoBufMut, S> {
+    pub(crate) msg: libc::msghdr,
+    pub(crate) addr: sockaddr_storage,
+    pub(crate) fd: SharedFd<S>,
+    pub(crate) buffer: T,
+    pub(crate) control: C,
+    pub(crate) slices: Vec<IoSliceMut>,
+    _p: PhantomPinned,
+}
+
+impl<T: IoVectoredBufMut, C: IoBufMut, S> RecvMsg<T, C, S> {
+    /// Create [`RecvMsgVectored`].
+    pub fn new(fd: SharedFd<S>, buffer: T, control: C) -> Self {
+        assert!(
+            control.as_buf_ptr().cast::<libc::cmsghdr>().is_aligned(),
+            "misaligned control message buffer"
+        );
+        Self {
+            addr: unsafe { std::mem::zeroed() },
+            msg: unsafe { std::mem::zeroed() },
+            fd,
+            buffer,
+            control,
+            slices: vec![],
+            _p: PhantomPinned,
+        }
+    }
+
+    pub(crate) fn set_msg(&mut self) {
```

Comment:
Either mark this as unsafe or accept a `Pin<&mut Self>`

Reply:
This is the same as https://github.com/compio-rs/compio/blob/e8f2a16e7d43a1e261dc929515a2ee0c1d921186/compio-driver/src/poll/op.rs#L591-L592

Reply:
Yeah I noticed. That's incorrect. Fix current ones first and I will handle the rest after merging this PR.